service/matching/handler.go

```diff
@@ -115,28 +117,38 @@ func (h *Handler) startRequestProfile(api string, scope int) tally.Stopwatch {
 
 // AddActivityTask - adds an activity task.
 func (h *Handler) AddActivityTask(ctx context.Context, addRequest *m.AddActivityTaskRequest) error {
+	startT := time.Now()
 	scope := metrics.MatchingAddActivityTaskScope
-	sw := h.startRequestProfile("AddActivityTask", scope)
-	defer sw.Stop()
 
 	if ok, _ := h.rateLimiter.TryConsume(1); !ok {
 		return h.handleErr(errMatchingHostThrottle, scope)
 	}
 
-	return h.handleErr(h.engine.AddActivityTask(addRequest), scope)
+	err, syncMatch := h.engine.AddActivityTask(addRequest)
```

Comment:
we can use the same scope, but create new metric name for this dispatch latency.